src/CST_derivings_generator.ml

```diff
@@ -0,0 +1,143 @@
+
+   This module contains all the visitors that can be generated using
+   {{: https://gitlab.inria.fr/fpottier/visitors }  the PPX visitors library },
+   namely: {ul {li iter} {li map} {li reduce} {li mapreduce} {li iter2}
+   {li map2} {li reduce2}}
+
+   The generated type for visitors on the whole concrete syntax tree can be
+   huge, which makes them unreadable and the files containing them ridiculously
+   big. They are therefore not generated and not shown here. We refer the user
+   to {{: http://gallium.inria.fr/~fpottier/visitors/manual.pdf } the
+   documentation of visitors }. *)
+
+(* The following disables odoc for the rest of the file. *)
+(**/**)
+ |} ;
+
+      intro_derivers = {|
+[@@deriving
+  visitors { variety = "iter";      name = "located_iter";      polymorphic = true },
+  visitors { variety = "map";       name = "located_map";       polymorphic = true },
+  visitors { variety = "reduce";    name = "located_reduce";    polymorphic = true },
+  visitors { variety = "mapreduce"; name = "located_mapreduce"; polymorphic = true },
+  visitors { variety = "iter2";     name = "located_iter2";     polymorphic = true },
+  visitors { variety = "map2";      name = "located_map2";      polymorphic = true },
+  visitors { variety = "reduce2";   name = "located_reduce2";   polymorphic = true }
+]
+        |} ;
+
+      cst_derivers = {|
+[@@deriving
+  visitors { variety = "iter";       ancestors = ["located_iter"];      nude = true },
+  visitors { variety = "map";        ancestors = ["located_map"];       nude = true },
+  visitors { variety = "reduce";     ancestors = ["located_reduce"];    nude = true },
+  visitors { variety = "mapreduce";  ancestors = ["located_mapreduce"]; nude = true },
+  visitors { variety = "iter2";      ancestors = ["located_iter2"];     nude = true },
+  visitors { variety = "map2";       ancestors = ["located_map2"];      nude = true },
+  visitors { variety = "reduce2";    ancestors = ["located_reduce2"];   nude = true }
+]
+        |}
+    }
+
+  | arg ->
+    Format.eprintf "I do not know what do with my first argument, `%s`.@." arg;
+    exit 2
+
+  | exception Invalid_argument _ ->
+    Format.eprintf "I expect a first argument.@.";
+    exit 2
+
+(* Get the content of the CST.mli file. *)
+
+let file_content =
+  let fname = "CST.mli" in
+  let ichan = open_in fname in
+  let buflen = 1024 in
+  let out = Buffer.create buflen in
+  let buf = Bytes.create buflen in
+  let rec read () =
+    match input ichan buf 0 buflen with
+    | 0 -> ()
+    | n -> Buffer.add_subbytes out buf 0 n; read ()
+  in
+  read ();
+  close_in ichan;
+  Buffer.contents out
+
+(* Add type equalities to all type definitions. *)
+
+let type_foo_eq_regexp =           (* this regular expression matches: *)
+  "\\(type\\|and\\)\\([^=\n]* \\)" (* "type" or "and" followed by anything that ends by a space *)
+  ^ "\\([^ =\n]+\\) *="            (* anything that looks like a type identifier, followed by = *)
```

Comment:
I see that you have answered my previous comment here by under specifying the language for type identifier. This is OK and leads to clearer code in my opinion.

Reply:
I also think this is a better way to do things indeed. But we can bring this regexp back to Shell if we decide we prefer the Shell approach of #126 over the OCaml approach of this PR.